File: app/src/main/java/com/animation/master/splash/SplashActivity.java
package com.animation.master.splash;

import android.animation.Animator;
import android.animation.AnimatorListenerAdapter;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import com.animation.master.AppNavigator;
import com.animation.master.R;
import com.animation.master.animation.RevealAnimation;
import com.animation.master.databinding.ActivitySplashBinding;

public class SplashActivity extends AppCompatActivity {

    private static final int DELAY = 1000;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final ActivitySplashBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_splash);

        binding.mainLayout.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
            @Override
            public void onLayoutChange(View v, int left, int top, int right, int bottom, int oldLeft, int oldTop, int oldRight, int oldBottom) {
                v.removeOnLayoutChangeListener(this);

                int cx = (left + right) / 2;
                int cy = (top + bottom) / 2;
                float radius = Math.max(binding.mainLayout.getWidth(), binding.mainLayout.getHeight()) * 2.0f;

                Animator reveal = RevealAnimation.getInstance()
                                                 .setCx(cx)
                                                 .setCy(cy)
                                                 .setStartRadius(0f)
                                                 .setEndRadius(radius)
                                                 .setDuration(1000)
                                                 .start(binding.mainLayout);

                if (reveal != null) {
                    reveal.addListener(new AnimatorListenerAdapter() {
                        @Override
                        public void onAnimationEnd(Animator animation) {
                            startMainActivity();
                        }
                    });
                }
            }
        });

    }

    private void startMainActivity() {
        try {
            Thread.sleep(DELAY);
        } catch (InterruptedException e) {
            e.printStackTrace();
        } finally {
            AppNavigator.navigateToDashboard(this);
        }
    }
}

File: app/src/main/java/com/animation/master/splash/SplashActivity.kt
package com.animation.master.splash

import android.animation.Animator
import android.animation.AnimatorListenerAdapter
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import com.animation.master.AppNavigator
import com.animation.master.R
import com.animation.master.animation.RevealAnimation
import com.animation.master.databinding.ActivitySplashBinding

class SplashActivity : AppCompatActivity() {

    companion object {
        private const val DELAY = 1000
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = DataBindingUtil.setContentView<ActivitySplashBinding>(this, R.layout.activity_splash)

        binding.mainLayout.addOnLayoutChangeListener(object : View.OnLayoutChangeListener {
            override fun onLayoutChange(v: View, left: Int, top: Int, right: Int, bottom: Int, oldLeft: Int, oldTop: Int, oldRight: Int, oldBottom: Int) {
                v.removeOnLayoutChangeListener(this)

                val cx = (left + right) / 2
                val cy = (top + bottom) / 2
                val radius = Math.max(binding.mainLayout.width, binding.mainLayout.height) * 2.0f

                val reveal = RevealAnimation.getInstance()
                    .setCx(cx)
                    .setCy(cy)
                    .setStartRadius(0f)
                    .setEndRadius(radius)
                    .setDuration(1000)
                    .start(binding.mainLayout)

                reveal?.addListener(object : AnimatorListenerAdapter() {
                    override fun onAnimationEnd(animation: Animator) {
                        startMainActivity()
                    }
                })
            }
        })

    }

    private fun startMainActivity() {
        try {
            Thread.sleep(DELAY.toLong())
        } catch (e: InterruptedException) {
            e.printStackTrace()
        } finally {
            AppNavigator.navigateToDashboard(this)
        }
    }
}
